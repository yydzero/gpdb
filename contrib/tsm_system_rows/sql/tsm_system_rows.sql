CREATE EXTENSION tsm_system_rows;

CREATE TABLE test_tablesample (id int, name text) WITH (fillfactor=10); -- force smaller pages so we don't have to load too much data to get multiple pages

INSERT INTO test_tablesample SELECT i, repeat(i::text, 1000) FROM generate_series(0, 30) s(i) ORDER BY i;
ANALYZE test_tablesample;

SELECT count(*) FROM test_tablesample TABLESAMPLE system_rows (1000);
SELECT id FROM test_tablesample TABLESAMPLE system_rows (8) REPEATABLE (5432);

EXPLAIN SELECT id FROM test_tablesample TABLESAMPLE system_rows (20) REPEATABLE (10);

-- done
DROP TABLE test_tablesample CASCADE;
