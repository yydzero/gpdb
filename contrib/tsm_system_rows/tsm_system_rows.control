# tsm_system_rows extension
comment = 'SYSTEM TABLESAMPLE method which accepts number rows as a limit'
default_version = '1.0'
module_pathname = '$libdir/tsm_system_rows'
relocatable = true
