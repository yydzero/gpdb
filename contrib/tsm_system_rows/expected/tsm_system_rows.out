CREATE EXTENSION tsm_system_rows;
CREATE TABLE test_tablesample (id int, name text) WITH (fillfactor=10); -- force smaller pages so we don't have to load too much data to get multiple pages
INSERT INTO test_tablesample SELECT i, repeat(i::text, 1000) FROM generate_series(0, 30) s(i) ORDER BY i;
ANALYZE test_tablesample;
SELECT count(*) FROM test_tablesample TABLESAMPLE system_rows (1000);
 count 
-------
    31
(1 row)

SELECT id FROM test_tablesample TABLESAMPLE system_rows (8) REPEATABLE (5432);
 id 
----
  7
 14
 21
 28
  4
 11
 18
 25
(8 rows)

EXPLAIN SELECT id FROM test_tablesample TABLESAMPLE system_rows (20) REPEATABLE (10);
                                    QUERY PLAN                                     
-----------------------------------------------------------------------------------
 Sample Scan (system_rows) on test_tablesample  (cost=0.00..80.20 rows=20 width=4)
(1 row)

-- done
DROP TABLE test_tablesample CASCADE;
